csharp-machine-learning/ProbabilisticProgrammingInCsharpWithInferNET/ProbabilisticProgrammingInCsharpWithInferNET/CoinTossEngine.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace ProbabilisticProgrammingInCsharpWithInferNET;
+public class CoinTossEngine
+{
+    private bool IsHead { get { return new Random().NextDouble() > 0.5; } }
```

Comment:
Use `Random.Shared.NextDouble()` instead of initializing a new `Random` instance